frontend/model/contracts/voting/rules.js

```diff
@@ -45,20 +59,22 @@ const rules = {
   },
   [RULE_DISAGREEMENT]: function (state, proposalType, votes) {
     votes = Object.values(votes)
-    const minimumMax = proposalType === PROPOSAL_REMOVE_MEMBER ? 2 : 0
-    const threshold = Math.max(state.settings.proposals[proposalType].ruleSettings[RULE_DISAGREEMENT].threshold, minimumMax)
+    const population = getPopulation(state)
+    const minimumMax = proposalType === PROPOSAL_REMOVE_MEMBER ? 2 : 1
+    const thresholdOriginal = Math.max(state.settings.proposals[proposalType].ruleSettings[RULE_DISAGREEMENT].threshold, minimumMax)
+    const threshold = getThresholdAdjusted(RULE_DISAGREEMENT, thresholdOriginal, population)
     const totalFor = votes.filter(x => x === VOTE_FOR).length
     const totalAgainst = votes.filter(x => x === VOTE_AGAINST).length
-    const population = Object.keys(state.profiles).length
     const turnout = votes.length
     const absent = population - turnout
+
     console.debug(`votingRule ${RULE_DISAGREEMENT} for ${proposalType}:`, { totalFor, totalAgainst, threshold, turnout, population, absent })
     if (totalAgainst >= threshold) {
       return VOTE_AGAINST
     }
     // consider proposal passed if more vote for it than against it and there aren't
     // enough votes left to tip the scales past the threshold
-    return totalAgainst + absent < threshold && totalFor > totalAgainst ? VOTE_FOR : VOTE_UNDECIDED
```

Comment:
Why was `&& totalFor > totalAgainst` removed?

I agree it's probably incorrect. Probably should check to make sure at least someone voted for it, so should be `&& totalFor > 0` is my thinking.

That's not quite enough to close these bugs but it's an improvement.

- https://github.com/okTurtles/group-income-simple/issues/876
- https://github.com/okTurtles/group-income-simple/issues/877

Reply:
I removed this while writing the tests. See the set `RULE_DISAGREEMENT - 4 - 5 members`, test `2VF vs 3VA returns for`. With the previous code, the proposal would return `VOTE_UNDECIDED `, but in reality, it should return `VOTE_FOR` because it's needed more `totalAgainst` than `totalFor` votes.

Why do we need to check if at least some voted for it?

Regarding the issue #876, I didn't understand how to fix the bug. I understand the problem but I don't know what changes to make taking in account the expire date. We don't even have the code to automatically close expired proposals, yet, right? So, maybe that should be done in another PR.

Regarding the issue #877, I prefer to leave it to a future PR. Probably at the same time as #878.

Reply:
> Why do we need to check if at least some voted for it?

Well that's an interesting question... should a proposal pass if no one is in favor of it...? But I do see your point. Purely with the way the disagreement rule is described, it is implied that a proposal can pass with no one voting explicitly in favor of it.

> Regarding #876, I didn't understand how to fix the bug. I understand the problem but I don't know what changes to make taking in account the expire date. We don't even have the code to automatically close expired proposals, yet, right? So, maybe that should be done in another PR.

Agreed, let's leave for another PR.

> Regarding #877, I prefer to leave it to a future PR. Probably at the same time as #878.

👍